src/main/java/org/openstreetmap/atlas/checks/distributed/ShardedIntegrityChecksSparkJob.java

```diff
@@ -0,0 +1,435 @@
+        final String input = Optional.ofNullable(input(commandMap)).orElse(atlasDirectory);
+
+        // Gather arguments
+        final String output = output(commandMap);
+        @SuppressWarnings("unchecked")
+        final Set<OutputFormats> outputFormats = (Set<OutputFormats>) commandMap
+                .get(OUTPUT_FORMATS);
+        final StringList countries = StringList.split((String) commandMap.get(COUNTRIES),
+                CommonConstants.COMMA);
+        @SuppressWarnings("unchecked")
+        final Optional<List<String>> checkFilter = (Optional<List<String>>) commandMap
+                .getOption(CHECK_FILTER);
+
+        final Configuration checksConfiguration = new MergedConfiguration(Stream
+                .concat(Stream.of(ConfigurationResolver.loadConfiguration(commandMap,
+                        CONFIGURATION_FILES, CONFIGURATION_JSON)),
+                        Stream.of(checkFilter
+                                .<Configuration> map(whitelist -> new StandardConfiguration(
+                                        "WhiteListConfiguration",
+                                        Collections.singletonMap(
+                                                "CheckResourceLoader.checks.whitelist", whitelist)))
+                                .orElse(ConfigurationResolver.emptyConfiguration())))
+                .collect(Collectors.toList()));
+
+        final Map<String, String> sparkContext = configurationMap();
+
+        // File loading helpers
+        final AtlasFilePathResolver resolver = new AtlasFilePathResolver(checksConfiguration);
+        final SparkFileHelper fileHelper = new SparkFileHelper(sparkContext);
+        final CheckResourceLoader checkLoader = new CheckResourceLoader(checksConfiguration);
+
+        // Spark storage argument
+        final StorageLevel storageLevel = Optional
+                .ofNullable(commandMap.get(SPARK_STORAGE_DISK_ONLY)).orElse("false").equals("true")
+                        ? StorageLevel.DISK_ONLY()
+                        : StorageLevel.MEMORY_AND_DISK();
+
+        // Get sharding
+        @SuppressWarnings("unchecked")
+        final Optional<String> alternateShardingFile = (Optional<String>) commandMap
+                .getOption(SHARDING);
+        final String shardingPathInAtlas = "dynamic@"
+                + SparkFileHelper.combine(input, ATLAS_SHARDING_FILE);
+        final String shardingFilePath = alternateShardingFile.orElse(shardingPathInAtlas);
+        final Sharding sharding = AtlasSharding.forString(shardingFilePath,
+                this.configurationMap());
+        final Broadcast<Sharding> shardingBroadcast = this.getContext().broadcast(sharding);
+        final Distance distanceToLoadShards = (Distance) commandMap.get(EXPANSION_DISTANCE);
+
+        // Check inputs
+        if (countries.isEmpty())
+        {
+            logger.error("No countries found to run");
+            return;
```

Comment:
One of @seancoulter's comments made me realize that there are a lot of soft failures used in the start method to short circuit execution of the new spark job. These were added in the original PR, so I had just left them alone. I am now wondering if it might be better to throw exceptions, to make it more evident why the job did not do anything. Does anyone have an opinion on this?